web/server/codechecker_server/api/report_server.py

```diff
@@ -1465,16 +1501,29 @@ def getRunResults(self, run_ids, limit, offset, sort_types,
                     report_ids = [r[1] for r in query_result]
                     report_details = get_report_details(session, report_ids)
 
+                # Earlier file table was joined to the query of reports.
+                # However, that created an SQL strategy in PostgreSQL which
+                # resulted in timeout. Based on heuristics the query strategy
+                # (which not only depends on the query statement but the table
+                # size and many other things) may contain an inner loop on
+                # "reports" table which is one of the largest tables. This
+                # separate query of file table results a query strategy for the
+                # previous query which doesn't contain such an inner loop. See
+                # EXPLAIN SELECT columns FROM ...
+                file_ids = set(map(lambda r: r[2], query_result))
+                all_files = dict(session.query(File.id, File.filepath)
+                                 .filter(File.id.in_(file_ids)).all())
```

Comment:
In my previous comment I mentioned to use our `chunk` util function because the number of variables which can be inside an `IN` statement is limited in case of SQLite (default: `SQLITE_MAX_VARIABLE_NUMBER=999`, https://www.sqlite.org/limits.html#max_variable_number). So if your database contains more files this query will fail.